src/server/service/kv.rs

```diff
@@ -52,6 +57,316 @@ pub struct Service<T: RaftStoreRouter + 'static, E: Engine, L: LockMgr> {
     snap_scheduler: Scheduler<SnapTask>,
 
     thread_load: Arc<ThreadLoad>,
+
+    // timeout set to 0 means batch is collected without crossing commands
+    minibatch_wait_millis: u64,
+
+    timer_pool: Arc<Mutex<ThreadPool>>,
+}
+
+#[derive(Hash, PartialEq, Eq, Debug)]
+struct RegionVerId {
+    region: u64,
+    epoch: u64,
+}
+
+struct BatchTimer {
+    timeout: u64,
+    timestamp: Instant,
+}
+
+impl BatchTimer {
+    fn new(timeout: u64) -> Self {
+        BatchTimer {
+            timeout,
+            timestamp: Instant::now(),
+        }
+    }
+
+    fn disabled(&self) -> bool {
+        self.timeout == 0
+    }
+
+    fn is_timeout(&self, now: Instant) -> bool {
+        now - self.timestamp >= Duration::from_millis(self.timeout)
+    }
+
+    fn set_now(&mut self, now: Instant) {
+        self.timestamp = now;
+    }
+}
+
+trait Batcher<E: Engine, L: LockMgr> {
```

Comment:
What is `Batcher`? What's the purpose of the trait? And what's the expected way to use it?